src/elements/content-sharing/types.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+import type { ItemType } from '../../common/types/core';
+import type { item, sharedLinkType } from '../../features/unified-share-modal/flowTypes';
+
+// "SLS" denotes values that are used in the Shared Link Settings modal
+type ContentSharingEnterpriseDataType = {
+    enterpriseName: string,
+    serverURL: string, // SLS
+};
+
+export type ContentSharingUserDataType = {
+    id: string,
+    userEnterpriseData: ContentSharingEnterpriseDataType,
+};
+
+export type ContentSharingSharedLinkType =
+    | { canInvite: boolean }
+    | (sharedLinkType &
+          ContentSharingEnterpriseDataType & {
+              canChangeDownload: boolean, // SLS
+              canChangePassword: boolean, // SLS
+              canChangeVanityName: boolean, // SLS
+              canInvite: boolean,
+              directLink: string, // SLS
+              isDirectLinkAvailable: boolean, // SLS
+              isDownloadAvailable: boolean, // SLS
+              isDownloadEnabled: boolean, // SLS
+              isPasswordAvailable: boolean, // SLS
+              isPasswordEnabled: boolean, // SLS
+              vanityName: string,
+          });
+
+export type ContentSharingItemDataType = {
+    item: item,
+    sharedLink: ContentSharingSharedLinkType,
+};
+
+export type ContentSharingItemAPIResponse = {
+    allowed_invitee_roles: Array<string>,
+    description: string,
+    etag: string,
+    extension: string,
+    id: string,
+    name: string,
+    permissions: {
```

Comment:
Should this share permissions with BoxItem? It appears to be missing some new permissions added to annotations, for example.